onadata/libs/throttle.py

```diff
@@ -4,7 +4,19 @@
 
 from django.conf import settings
 
-from rest_framework.throttling import SimpleRateThrottle
+from rest_framework.throttling import SimpleRateThrottle, ScopedRateThrottle
+
+
+class CustomScopedRateThrottle(ScopedRateThrottle):
+    """
+    Custom throttling for fair throttling for anonymous users sharing IP
+    """
+
+    def get_cache_key(self, request, view):
+        if request.user and request.user.is_authenticated:
```

Comment:
Should we do
```
def get_cache_key(self, request, view):
    xff = request.META.get('HTTP_X_FORWARDED_FOR')

    if (request.user and request.user.is_authenticated) or xff is not None:
        return super().get_cache_key(request, view)

    return f'throttle_{self.scope}_{request.path}'

```

Reply:
I'm not sure about this since [get_ident](https://github.com/encode/django-rest-framework/blob/master/rest_framework/throttling.py#L197) is only called when request.user is `None`

I think we should use get_ident to make the cache key more speicif